packages/data-point/lib/reducer-types/resolve.js

```diff
@@ -45,26 +46,67 @@ function getResolveFunction (reducer) {
  * @param {Object} manager
  * @param {Accumulator} accumulator
  * @param {Reducer} reducer
+ * @param {Array|String|Number} key - id for reducer stack traces if an error is thrown
  * @returns {Promise<Accumulator>}
  */
-function resolveReducer (manager, accumulator, reducer) {
+function resolveReducer (manager, accumulator, reducer, key) {
   // this conditional is here because BaseEntity#resolve
   // does not check that lifecycle methods are defined
   // before trying to resolve them
   if (!reducer) {
     return Promise.resolve(accumulator)
   }
 
-  const resolve = getResolveFunction(reducer)
-  // NOTE: recursive call
-  const result = resolve(manager, resolveReducer, accumulator, reducer)
+  // storing this in case we need it for the catch block, since we
+  // can't trust it won't be overwritten in the accumulator object
+  // although it could still be modified by reference if it's an object
+  const value = accumulator.value
+  const result = Promise.try(() => getResolveFunction(reducer))
+    // NOTE: recursive call
+    .then(resolve => resolve(manager, resolveReducer, accumulator, reducer))
+
   if (hasDefault(reducer)) {
     const _default = reducer[DEFAULT_VALUE].value
     const resolveDefault = reducers.ReducerDefault.resolve
     return result.then(acc => resolveDefault(acc, _default))
   }
 
-  return result
+  return result.catch(error => onResolveMalfunction(reducer, key, value, error))
 }
 
 module.exports.resolve = resolveReducer
+
+/**
+ * @param {Reducer} reducer
+ * @param {Array|String|Number} key
+ * @param {*} value
+ * @param {Error} error
+ * @throws the given error with _value and _stack properties attached
+ */
+function onResolveMalfunction (reducer, key, value, error) {
+  if (!error.hasOwnProperty('_value')) {
+    error._value = value
+  }
+
+  let stack
+  if (typeof key === 'undefined') {
+    stack = []
+  } else {
+    stack = castArray(key)
+  }
+
+  if (reducer.type === 'ReducerFunction') {
```

Comment:
Might this be better as a `switch`? I don't know what other types we might have a case for, but at a certain point `if..else if..else` gets cumbersome to read.

Reply:
We probably won't need more conditions here, since I don't think we'll be adding many more reducer types. Honestly I just really dislike switch syntax...

Reply:
> Honestly I just really dislike switch syntax...

Ah ha!!!! 😄 

But for real, if we're not going to extend it then a switch isn't really necessary I guess.

Reply:
Happy to change it though if there's a consensus...

Reply:
It might be premature to change it now if we only have to account for these two scenarios.